lib/ui/text.dart

```diff
@@ -1489,6 +1490,24 @@ class ParagraphBuilder extends NativeFieldWrapperClass2 {
   }
   String _addText(String text) native 'ParagraphBuilder_addText';
 
+  /// Adds a space for a widget to the paragraph.
+  ///
+  /// The paragraph will contain a rectangular space with no text of the size
+  /// specified.
+  ///
+  /// The [width] and [height] parameters specify the size of the rectangle, and
+  /// [baseline] indicates the distance of the alphabetic baseline from the top of
+  /// of the rectangle. This determines how the rectangle is aligned to the
+  /// surrounding text.
+  ///
+  /// For example, to make a 50x50 rectangle that sits completely on top of the
+  /// alphabetic baseline, pass in `addWidget(50, 50, 50)`. For a box that is
+  /// vertically centered on the alphabetic baseline, pass in `addWidget(50, 50, 25);`.
+  void addWidget(double width, double height, double baseline) {
```

Comment:
"Widget" is probably the wrong term here; this layer doesn't know about widgets.
What do you think of "Object", "Box", or "Placeholder"?

Reply:
Ahh, I'll give it some thought and rename everything with widget in it!

Reply:
So "Object" is ambiguous with OOP objects, "Box/Rect" is a phrase we currently use to describe bounding boxes around characters, so it would be subpar to add a new concept with the same name.

"Placeholder" seems pretty reasonable, but I wonder if that can be something shorter or more descriptive or concise.